### PR TITLE
[CELEBORN-1601][FOLLOWUP] Refine the RESTful apis for revise lost shuffles

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?

1. `GET /api/v1/applications/deleteApps`  -> `DELETE /api/v1/applications`

2. `GET /api/v1/applications/reviseLostShuffles`  -> `POST /api/v1/applications/revise_lost_shuffles`

### Why are the changes needed?
Followup for https://github.com/apache/celeborn/pull/2746

### Does this PR introduce _any_ user-facing change?

No, these APIs has not been released yet.

### How was this patch tested?
GA.

### CODE REVIEW

**Comment by SteNicholas** [comment]
@turboFei, is there any tool that could verify the consistency between generated openapi-client code and rest api model?

File: master/src/main/scala/org/apache/celeborn/service/deploy/master/http/api/v1/ApplicationResource.scala [hunk 1]
```diff
@@ -55,6 +54,19 @@ class ApplicationResource extends ApiRequestContext {
...
   }
 
+  @ApiResponse(
+    responseCode = "200",
+    content = Array(new Content(
+      mediaType = MediaType.APPLICATION_JSON,
+      schema = new Schema(implementation = classOf[HandleResponse]))),
+    description = "Delete resource of an app")
+  @DELETE
+  def deleteApp(request: DeleteAppsRequest): HandleResponse = {
```
Comment (SteNicholas) [inline master/src/main/scala/org/apache/celeborn/service/deploy/master/http/api/v1/ApplicationResource.scala:64]:
```suggestion
  def deleteApps(request: DeleteAppsRequest): HandleResponse = {
```

File: master/src/main/scala/org/apache/celeborn/service/deploy/master/http/api/v1/ApplicationResource.scala [hunk 2]
```diff
@@ -96,41 +108,18 @@ class ApplicationResource extends ApiRequestContext {
     new HostnamesResponse().hostnames(statusSystem.hostnameSet.asScala.toSeq.asJava)
   }
 
-  @Path("/reviseLostShuffles")
+  @Path("/revise_lost_shuffles")
```
Comment (SteNicholas) [inline master/src/main/scala/org/apache/celeborn/service/deploy/master/http/api/v1/ApplicationResource.scala:111]: Is there any standard of path pattern? If yes, could you add some description in `CONTRIBUTING.md` to tell contributor this standard.

File: master/src/main/scala/org/apache/celeborn/service/deploy/master/http/api/v1/ApplicationResource.scala (hunk 1 above)
Comment (turboFei) [reply master/src/main/scala/org/apache/celeborn/service/deploy/master/http/api/v1/ApplicationResource.scala]: done

File: master/src/main/scala/org/apache/celeborn/service/deploy/master/http/api/v1/ApplicationResource.scala (hunk 2 above)
Comment (turboFei) [reply master/src/main/scala/org/apache/celeborn/service/deploy/master/http/api/v1/ApplicationResource.scala:111]: update the readme

Comment (turboFei) [reply master/src/main/scala/org/apache/celeborn/service/deploy/master/http/api/v1/ApplicationResource.scala:111]: image

**Comment by turboFei** [comment]
> @turboFei, is there any tool that could verify the consistency between generated openapi-client code and rest api model?

Not yet now, maybe we can followup it to diff the generated RESTful modules in GA.

**Comment by turboFei** [comment]
> consistency between generated openapi-client code



> > @turboFei, is there any tool that could verify the consistency between generated openapi-client code and rest api model?
> 
> Not yet now, maybe we can followup it to diff the generated RESTful modules in GA.
> 
> Raised https://issues.apache.org/jira/browse/CELEBORN-1699 to track

Addressed in https://github.com/apache/celeborn/pull/2893

**Comment by turboFei** [comment]
Will move the ThreadStack part into https://github.com/apache/celeborn/pull/2893

**Comment by turboFei** [comment]
rebased the code.

File: cli/src/main/scala/org/apache/celeborn/cli/master/MasterSubcommandImpl.scala [hunk 3]
```diff
@@ -224,18 +225,39 @@ class MasterSubcommandImpl extends Runnable with MasterSubcommand {
   private[master] def runShowContainerInfo: ContainerInfo = defaultApi.getContainerInfo
 
   override private[master] def reviseLostShuffles: HandleResponse = {
+    if (StringUtils.isEmpty(commonOptions.apps)) {
```
Comment (SteNicholas) [inline cli/src/main/scala/org/apache/celeborn/cli/master/MasterSubcommandImpl.scala:228]: Could this condition combine with line 240?

Comment (turboFei) [reply cli/src/main/scala/org/apache/celeborn/cli/master/MasterSubcommandImpl.scala:228]: sure

**Review by SteNicholas — APPROVED** [review]
Only one minor comment.

File: cli/src/main/scala/org/apache/celeborn/cli/master/MasterSubcommandImpl.scala (hunk 3 above)
Comment (turboFei) [reply cli/src/main/scala/org/apache/celeborn/cli/master/MasterSubcommandImpl.scala]: thanks, done

**Review by FMX — APPROVED** [review]
Thanks. A modernized HTTP API implementation. LGTM.

**Review by RexXiong — APPROVED** [review]
:+1:  LGTM

**Pull request closed without merging**